pkg/executor/utils.go

```diff
@@ -119,3 +121,70 @@ func encodePassword(u *ast.UserSpec, authPlugin *extension.AuthPlugin) (string,
 	}
 	return u.EncodedPassword()
 }
+
+var taskPool = sync.Pool{
+	New: func() any { return &gotask{} },
+}
+
+type gotask struct {
+	f    func()
+	next *gotask
+}
+
+type gopool struct {
+	lock sync.Mutex
+	head *gotask
```

Comment:
How about using a lock free queue to store the pending task pointers, so the `Mutex` could be avoided?

Reply:
Here gopool is mainly for concurrency control, each IndexLookupExecutor has its own pool, that is, there are only about indexLookupConcurrency goroutines access the pool at most 2xNumOfTableTasks times. I think it's acceptable, no need to introduce additional complexity.